sos/plugins/cnv.py

```diff
@@ -0,0 +1,234 @@
+                self.add_cmd_output(["%s get pvc -o yaml " % oc_cmd + pvc])
+        self.add_cmd_output("%s get vm --all-namespaces" % oc_cmd)
+        vmis = self.get_cmd_output_now("%s get vmi --all-namespaces" % oc_cmd)
+
+        if self.get_option("vm"):
+            options = self.get_option("vm")
+            if isinstance(options, bool):
+                return
+
+            subcmds_vm = [
+                "describe vm",
+                "describe vmi",
+                "get vm -o yaml",
+                "get vmi -o yaml",
+            ]
+            node_pods = [
+                "%s get pods -l kubevirt.io=virt-handler "
+                "--all-namespaces -o wide" % oc_cmd,
+                "%s get pods -l app=multus "
+                "--all-namespaces -o wide" % oc_cmd,
+                "%s get pods -l app=ovs-cni "
+                "--all-namespaces -o wide" % oc_cmd,
+                "%s get pods -l app=ovs "
+                "--all-namespaces -o wide" % oc_cmd,
+                "%s get pods -l app=sdn "
+                "--all-namespaces -o wide" % oc_cmd,
+            ]
+
+            for vm in self.get_vm_ns(vmis, self.get_option("vm")):
+                for s in subcmds_vm:
+                    self.add_cmd_output(["%s " % oc_cmd + s + " " + vm])
+                getvmi = "%s get vmi " % oc_cmd + vm + " --show-labels"
+                lbl_ext = self.call_ext_prog(getvmi)
+                vmi_l = ""
+                if lbl_ext['status'] != 0:
+                    return
+                for line in lbl_ext['output'].splitlines():
+                    if line.split()[0] == vm.split()[0]:
+                        for l in line.split()[2].split(','):
+                            if vm.split()[0] in l:
+                                vmi_l = "-l " + l + " --all-namespaces -o wide"
+                p_ln = self.get_cmd_output_now("%s get pods " % oc_cmd + vmi_l)
+                for pod in self.get_resc_ns(p_ln):
+                    for s in subcmds_pods:
+                        self.add_cmd_output(["%s " % oc_cmd + s + " " + pod])
+                    exc = "%s " % oc_cmd + "exec " + pod + " -- virsh dumpxml "
+                    dmn = vm.split()[2] + "_" + vm.split()[0]
+                    self.add_cmd_output([exc + dmn])
+                node = self.get_node_info(p_ln)
+                vm_pods = []
+                for pod in node_pods:
+                    node_pod_ext = self.call_ext_prog(pod)
+                    if node_pod_ext['status'] == 0:
+                        for line in node_pod_ext['output'].splitlines():
+                            if line.startswith("NAMESPACE") \
+                                    or line.startswith("No resources found") \
+                                    or line.startswith("Error") \
+                                    or line.startswith("error"):
+                                pass
+                            else:
+                                if line.split()[7] == node:
```

Comment:
Ok, now I'm a bit confused at this point. We seem to running multiple `get pods` commands for those with specific cnv-related labels, and trying to extract a pod (yes? node?) name and see if it matches the node name we got from a  generic `get pods` that is looking for specific labels? 

Can you provide an example of what exactly we're comparing against? Because right now I don't follow why we need to jump through all these hoops instead of just being able to say "get all the pod names that are cnv resources with XYZ label" like we're able to do with other kube/origin resources.